src/service.rs

```diff
@@ -292,13 +318,7 @@ where
 
         match dial_future.await {
             Ok(value) => {
-                self.session_event_sender
-                    .send(SessionEvent::DialStart {
-                        remote_address: value.0,
-                        stream: value.1,
-                    })
-                    .await
-                    .map_err(|_| TransportErrorKind::Io(io::ErrorKind::BrokenPipe.into()))?;
+                self.handshake(value.1, SessionType::Outbound, value.0, None);
```

Comment:
there are several `value.0`, `value.1` usage in current code, I think it's a good chance to refactor these to :

```
match ... {
  Ok((address, incoming))
}
```